### PR TITLE
Issue / 6707 Improve Inline Error Notice & Retry UX

### PR DESCRIPTION
**Pull request by 10upsimon**
## Summary

Addresses issue:

- #6707

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/ErrorNotice.js [hunk 1]
```diff
@@ -63,7 +64,16 @@ export default function ErrorNotice( {
...
-	const shouldDisplayRetry = isErrorRetryable( error, selectorData );
+	const shouldDisplayRetry =
+		hasButton && isErrorRetryable( error, selectorData );
+
+	// Append try again messaging if no retry button is present.
+	if ( ! hasButton ) {
+		message = `${ message }. ${ __(
+			'Please try again.',
+			'google-site-kit'
+		) }`;
```
Comment (tofumatt) [inline assets/js/components/ErrorNotice.js:75]: We shouldn't be using punctuation inside strings like this where we combine user-facing text if we can help it. This should all be localisable, eg.

```suggestion
		message = sprintf( __(
			'%s. Please try again.',
			'google-site-kit'
		), message );
```

File: assets/js/components/ErrorNotice.js [hunk 2]
```diff
@@ -63,7 +64,16 @@ export default function ErrorNotice( {
 		return null;
 	}
 
-	const shouldDisplayRetry = isErrorRetryable( error, selectorData );
+	const shouldDisplayRetry =
+		hasButton && isErrorRetryable( error, selectorData );
+
+	// Append try again messaging if no retry button is present.
```
Comment (tofumatt) [inline assets/js/components/ErrorNotice.js:70]:
```suggestion
	// Append "Try again" messaging if no retry button is present.
```

**Review by tofumatt — CHANGES_REQUESTED** [review]
Just a small tweak to the text and this should be good 🙂

File: assets/js/components/ErrorNotice.js (hunk 1 above)
Comment (10upsimon) [reply assets/js/components/ErrorNotice.js:75]: Thanks @tofumatt this makes sense, I am going to commit your suggestion.

**Comment by 10upsimon** [comment]
@tofumatt back to you for another round of review, appreciate your feedback

**Review by tofumatt — APPROVED** [review]

**Pull request merged**